components/brave_ads/core/internal/ads/serving/choose/eligible_ads_predictor_util.h

```diff
@@ -159,6 +165,44 @@ CreativeAdPredictorMap<T> ComputePredictorFeaturesAndScores(
   return creative_ad_predictors_with_features;
 }
 
+template <typename T>
+std::vector<int> ComputeVoteRegistry(
+    const std::vector<T>& creative_ads,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  DCHECK(!creative_ads.empty());
+
+  std::vector<int> vote_registry(creative_ads.size());
+
+  for (const auto& text_embedding_html_event : text_embedding_html_events) {
+    std::vector<float> similarity_scores;
+
+    for (const auto& creative_ad : creative_ads) {
+      const ml::VectorData ad_embedding(creative_ad.embedding);
```

Comment:
What size in average `creative_ad.embedding` and `text_embedding_html_event.embedding` will have?

Reply:
should be 300-dimensional

Reply:
Can we please check memory usage before and after this change, and optimize where possible. Thanks

Reply:
yep, we definitely need to check memory consumption. do we have an upper limit for text_embedding_html_event size?

Reply:
right now it's tie to Griffin and the default max history size is 10

Reply:
Do we plan to increase this value after launch of this feature, or for now do we expect the value to stay at 10?

Reply:
it's likely that we are going to increase this in the future

Reply:
We should now discuss the potential under bounds and run performance tests based on those.